sshd-core/src/main/java/org/apache/sshd/client/config/keys/ClientIdentityFileWatcher.java

```diff
@@ -98,6 +98,13 @@ protected KeyPair reloadClientIdentity(Path path) throws IOException, GeneralSec
                 }
                 return null;
             }
+            String ownerViolation = KeyUtils.validateStrictKeyFileOwner(path, IoUtils.EMPTY_LINK_OPTIONS);
+            if (ownerViolation != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("reloadClientIdentity({}) ignore due to owner violation: {}", path, ownerViolation);
+                }
```

Comment:
I prefer:
```java
if (GenericUtils.length(ownerViolation) > 0) {
    ...
}
```
I don't think "empty" owner are expected, and even if they are IMO they are the same as "no user".